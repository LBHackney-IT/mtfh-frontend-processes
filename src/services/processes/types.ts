export enum Trigger {
  CloseProcess = "CloseProcess",
  CancelProcess = "CancelProcess",
}
