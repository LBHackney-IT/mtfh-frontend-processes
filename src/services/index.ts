export { default as locale } from "./locale";
export { default as menu } from "./menu";
export * as processes from "./processes";
export * from "./config";
