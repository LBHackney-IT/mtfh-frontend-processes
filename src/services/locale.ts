import reactHtmlParser from "react-html-parser";

import { AssetAddress } from "@mtfh/common/lib/api/asset/v1";

const locale = {
  backButton: "Back",
  bookAppointment: "Book Appointment",
  next: "Next",
  cancel: "Cancel",
  cancelProcess: "Cancel process",
  change: "Change",
  views: {
    soleToJoint: {
      title: "Sole tenant requests a joint tenure",
      steps: {
        selectTenant: "Select person",
        checkEligibility: "Eligibility Checks",
        supportingDocuments: "Supporting documents",
        finish: "Finish",
        breachOfTenancy: "Breach of tenure check",
        requestDocuments: "Request Documents",
        reviewDocuments: "Review Documents",
        submitCase: "Submit case",
      },
      actions: {
        reassignCase: "Reassign case",
        cancelProcess: "Cancel process",
        caseActivityHistory: "Case activity history",
      },
    },
    selectTenants: {
      selectTenantHint:
        "This person will be asked for proof of relationship e.g. marriage or civil partnership certificate.",
      selectTenantLabel: "Who do you want to add as a joint tenant?",
      addToTenureText: "If the person you want to add is not listed you must first",
      addToTenureLink: "add them to the tenure",
      noHouseholdMembersOver18:
        "There are no household members over the age of 18 associated to this.",
    },
    processesMenu: {
      selectProcessDescription: () =>
        reactHtmlParser(
          "<p>Start a process by selecting a process category below. <br /> Where a process is not yet supported in <strong>Manage My Home</strong> these links will open the appropriate interim Google form.</p>",
        ),
    },
    checkEligibility: {
      autoCheckIntro:
        "Manage My Home has used tenure, person and property information to automatically check eligibility for this application.",
      passedChecks: "Passed automatic eligibilty checks",
      failedChecks: "Not eligible for a sole to joint tenure",
    },
    breachOfTenancy: {
      autoCheckIntro:
        "Manage My Home has used tenure, person and property information to automatically check eligibility for this application.",
      passedChecks: "Eligibility checks passed",
      failedChecks: "Not eligible for a sole to joint tenure",
    },
    reviewDocuments: {
      passedChecks: "Eligibility checks passed",
      documentsRequested: "Documents requested",
      checkSupportingDocumentsAppointment:
        "I need to make an appointment to check supporting documents",
      appointmentScheduled: "Office appointment scheduled",
      seenPhotographicId:
        "I confirm that an outcome letter has been sent to the resident",
      seenSecondId:
        "I confirm I have seen a second form of ID (does not have to be photographic)",
      isNotInImmigrationControl:
        "I confirm that the prospective tenant is not subject to immigration control under the Asylum And Immigration Act 1996",
      seenProofOfRelationship:
        "I confirm I have seen proof of relationship to the existing tenant",
      incomingTenantLivingInProperty:
        "I confirm I have seen 3 separate documents proving the proposed tenant has been living at the property for a minimum of 12 months",
    },
  },
  errors: {
    unableToFetchRecord: "There was a problem retrieving the record",
    unableToFetchRecordDescription:
      "Please try again. If the issue persists, please contact support.",
    unableToFindProcess: "Invalid Process",
    unableToFindProcessDescription: "We have been unable to find a process of this type.",
    unableToFindState: "Invalid State",
    unableToFindStateDescription: "We have been unable to process a state of this type.",
  },
  title: "Processes",
  loadingText: "Submitting...",
  components: {
    startProcess: {
      buttonLabel: "Start process",
      thirdPartyHeading: "Sharing information with third parties",
    },
    entitySummary: {
      tenurePaymentRef: "Tenure payment ref",
      address: (assetAddress: AssetAddress): string => {
        const {
          postPreamble,
          addressLine1,
          addressLine2,
          addressLine3,
          addressLine4,
          postCode,
        } = assetAddress;

        return [
          postPreamble,
          addressLine1,
          addressLine2,
          addressLine3,

          addressLine4,
          postCode,
        ]
          .filter((addressLine) => !!addressLine)
          .join(" ");
      },
    },
  },
};

export default locale;
