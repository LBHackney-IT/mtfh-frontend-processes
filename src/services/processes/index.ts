export * from "./sole-to-joint";
export * from "./change-of-name";
