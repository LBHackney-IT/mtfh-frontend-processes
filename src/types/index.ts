export type TargetType = "tenure" | "property" | "person";

export interface IStartProcess {
  thirdPartyCondition?: string;
  thirdPartyComponent?: React.FC;
  subHeading?: string;
  subComponent?: React.FC;
}

export interface IProcess {
  processName: string;
  title: string;
  targetType: TargetType;
  startProcess: IStartProcess;
  states: {
    [state: string]: {
      state: string;
      triggers: Record<string, string>;
    };
  };
}
