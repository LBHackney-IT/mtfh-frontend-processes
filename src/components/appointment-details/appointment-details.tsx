import { format, isPast } from "date-fns";

import { locale } from "../../services";

import { Process } from "@mtfh/common/lib/api/process/v1";
import { Box, LinkButton, StatusHeading, Text } from "@mtfh/common/lib/components";

interface AppointmentDetailsProps {
  process: Process;
  needAppointment: boolean;
  setNeedAppointment: any;
  options: {
    requestAppointmentTrigger: string;
    rescheduleAppointmentTrigger: string;
    appointmentRequestedState: string;
    appointmentRescheduledState: string;
    cancelProcess?: boolean;
  };
}

export const AppointmentDetails = ({
  process,
  needAppointment,
  setNeedAppointment,
  options,
}: AppointmentDetailsProps): JSX.Element => {
  const { currentState } = process;
  const formData = process.currentState.processData.formData as {
    appointmentDateTime: string;
  };

  return (
    <>
      {[options.appointmentRequestedState, options.appointmentRescheduledState].includes(
        currentState.state,
      ) && (
        <>
          {options.appointmentRescheduledState === process.currentState.state &&
            process.previousStates.map((process) => {
              if (process.state === options.appointmentRequestedState) {
                return (
                  <Box key={process.state}>
                    <StatusHeading
                      variant="base"
                      title={locale.components.appointment.missed}
                    />
                    <Text style={{ marginLeft: 60 }}>
                      Date:{" "}
                      {format(
                        new Date(process.processData.formData.appointmentDateTime),
                        "eeee do MMMM yyyy",
                      )}
                      <br />
                      Time:{" "}
                      {format(
                        new Date(process.processData.formData.appointmentDateTime),
                        "hh:mm aaa",
                      )}
                    </Text>
                  </Box>
                );
              }
              return null;
            })}

          <Box>
            <StatusHeading
              variant="base"
              title={locale.components.appointment.scheduled}
            />
            <Text style={{ marginLeft: 60 }}>
              Date: {format(new Date(formData.appointmentDateTime), "eeee do MMMM yyyy")}
              <br />
              Time: {format(new Date(formData.appointmentDateTime), "hh:mm aaa")}
            </Text>

            <LinkButton
              style={{ marginLeft: 60 }}
              onClick={() => setNeedAppointment(!needAppointment)}
            >
              {isPast(new Date(formData.appointmentDateTime))
                ? locale.reschedule
                : locale.change}
            </LinkButton>
            {options.cancelProcess &&
              options.appointmentRescheduledState === process.currentState.state &&
              isPast(new Date(formData.appointmentDateTime)) && (
                <LinkButton
                  style={{ marginLeft: 60 }}
                  onClick={() => setNeedAppointment(!needAppointment)}
                >
                  {locale.components.appointment.cancelProcess}
                </LinkButton>
              )}
          </Box>
        </>
      )}
    </>
  );
};
