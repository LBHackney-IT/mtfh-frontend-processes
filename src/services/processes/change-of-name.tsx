import { IProcess, IStartProcess } from "../../types";
import locale from "../locale";
import { Process, Trigger } from "./types";

import { Text } from "@mtfh/common/lib/components";

const startProcess: IStartProcess = {
  thirdPartyComponent: () => {
    return (
      <Text>
        By starting this process the tenant must agree that information relating to this
        application can be shared with third parties, for instance to perform a credit
        check.
      </Text>
    );
  },
  thirdPartyCondition:
    "I have explained to the tenant that their information will be shared with third parties.",
  subHeading: locale.supportingDocuments,
  subComponent: () => {
    return (
      <div>
        <Text>
          The applicant will be asked to provide documents as evidence to support their
          application.
        </Text>
      </div>
    );
  },
};

export const changeofname: IProcess = {
  processName: Process.ChangeOfName,
  targetType: "person",
  title: locale.views.changeofname.title,
  startProcess,
  states: {
    enterNewName: {
      state: "EnterNewName",
      status: "Awaiting tenant's new name",
      triggers: { enterNewName: "EnterNewName" },
    },
    nameSubmitted: {
      state: "NameSubmitted",
      status: "Awaiting supporting documents through DES",
      triggers: {
        requestDocumentsDES: Trigger.RequestDocumentsDES,
        requestDocumentsAppointment: Trigger.RequestDocumentsAppointment,
        cancelProcess: Trigger.CancelProcess,
      },
    },
    documentsRequestedDes: {
      state: "DocumentsRequestedDes",
      status: "Awaiting supporting documents through DES",
      triggers: {
        reviewDocuments: Trigger.ReviewDocuments,
        requestDocumentsAppointment: Trigger.RequestDocumentsAppointment,
        cancelProcess: Trigger.CancelProcess,
      },
    },
    documentsRequestedAppointment: {
      state: "DocumentsRequestedAppointment",
      status: "Awaiting supporting documents appointment",
      triggers: {
        reviewDocuments: Trigger.ReviewDocuments,
        rescheduleDocumentsAppointment: Trigger.RescheduleDocumentsAppointment,
        cancelProcess: Trigger.CancelProcess,
      },
    },
    documentsAppointmentRescheduled: {
      state: "DocumentsAppointmentRescheduled",
      status: "Awaiting supporting documents appointment",
      triggers: {
        reviewDocuments: Trigger.ReviewDocuments,
        rescheduleDocumentsAppointment: Trigger.RescheduleDocumentsAppointment,
        cancelProcess: Trigger.CancelProcess,
      },
    },
    documentChecksPassed: {
      state: "DocumentChecksPassed",
      status: "Submit for Tenure investigation",
      triggers: {
        submitApplication: Trigger.SubmitApplication,
        hOApproval: Trigger.HOApproval,
      },
    },
    applicationSubmitted: {
      state: "ApplicationSubmitted",
      status: "Awaiting Tenure investigation",
      triggers: { tenureInvestigation: Trigger.TenureInvestigation },
    },
    tenureInvestigationFailed: {
      state: "TenureInvestigationFailed",
      status: "Awaiting Housing Officer review",
      triggers: {
        hOApproval: Trigger.HOApproval,
        scheduleInterview: Trigger.ScheduleInterview,
      },
    },
    tenureInvestigationPassed: {
      state: "TenureInvestigationPassed",
      status: "Awaiting Housing Officer review",
      triggers: {
        hOApproval: Trigger.HOApproval,
        scheduleInterview: Trigger.ScheduleInterview,
      },
    },
    tenureInvestigationPassedWithInt: {
      state: "TenureInvestigationPassedWithInt",
      status: "Awaiting Housing Officer review",
      triggers: {
        hOApproval: Trigger.HOApproval,
        scheduleInterview: Trigger.ScheduleInterview,
      },
    },
    interviewScheduled: {
      state: "InterviewScheduled",
      status: "Awaiting Tenure Investigation Interview",
      triggers: {
        hOApproval: Trigger.HOApproval,
        rescheduleInterview: Trigger.RescheduleInterview,
        cancelProcess: Trigger.CancelProcess,
      },
    },
    interviewRescheduled: {
      state: "InterviewRescheduled",
      status: "Awaiting Tenure Investigation Interview",
      triggers: { hOApproval: Trigger.HOApproval, cancelProcess: Trigger.CancelProcess },
    },
    hoApprovalFailed: {
      state: "HOApprovalFailed",
      status: "Process Closed",
      triggers: { cancelProcess: Trigger.CancelProcess },
    },
    hoApprovalPassed: {
      state: "HOApprovalPassed",
      status: "Schedule new tenancy signing appointment",
      triggers: {
        scheduleTenureAppointment: Trigger.ScheduleTenureAppointment,
        updateName: Trigger.UpdateName,
        cancelProcess: Trigger.CancelProcess,
      },
    },
    tenureAppointmentScheduled: {
      state: "TenureAppointmentScheduled",
      status: "Awaiting new tenancy signing appointment",
      triggers: {
        rescheduleTenureAppointment: Trigger.RescheduleTenureAppointment,
        updateTenure: Trigger.UpdateName,
        cancelProcess: Trigger.CancelProcess,
      },
    },
    tenureAppointmentRescheduled: {
      state: "TenureAppointmentRescheduled",
      status: "Awaiting new tenancy signing appointment",
      triggers: {
        rescheduleTenureAppointment: Trigger.RescheduleTenureAppointment,
        updateTenure: Trigger.UpdateName,
        cancelProcess: Trigger.CancelProcess,
      },
    },
    nameUpdated: {
      state: "NameUpdated",
      status: "Process completed",
      triggers: {},
    },
    processClosed: {
      state: "ProcessClosed",
      status: "Process Closed",
      triggers: {},
    },
    processCancelled: {
      state: "ProcessCancelled",
      status: "Process Cancelled",
      triggers: {},
    },
  },
};
