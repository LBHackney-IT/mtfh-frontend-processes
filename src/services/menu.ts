import * as processes from "./processes";

import { Asset } from "@mtfh/common/lib/api/asset/v1/types";
import { Tenure } from "@mtfh/common/lib/api/tenure/v1/types";

interface MenuProps {
  label: string;
  link: string;
  processes?: { label: string; link: string; targetType: string }[];
  getPplQuery?: (context: any) => string;
}

const getProcessLink = (processName: string) => {
  const process = processes[processName];
  return {
    targetType: process.targetType,
    label: process.title,
    link: `/processes/${process.processName}/start`,
  };
};

const getPplQueryForRequestALegalReferral = ({ tenure, asset }): string => {
  if (!tenure || !asset) {
    return "";
  }
  const {
    householdMembers,
    tenuredAsset: { propertyReference },
  } = tenure as Tenure;
  const {
    assetAddress: { addressLine1, postCode },
  } = asset as Asset;
  const tenantNames = householdMembers
    .filter((member) => member.isResponsible)
    .map((member) => member.fullName)
    .join(", ");
  const address = addressLine1;
  const query = `usp=pp_url&entry.1699199605=${tenantNames}&entry.2086190845=${address}&entry.1956036523=${postCode}&entry.1140527432=${propertyReference}`;
  const encodedQuery = encodeURI(query);
  return encodedQuery;
};

const getPplQueryTenancyChange = ({ tenure }): string => {
  if (!tenure) {
    return "";
  }
  const {
    householdMembers,
    paymentReference,
    tenuredAsset: { fullAddress, propertyReference },
  } = tenure as Tenure;

  const tenant = householdMembers.find((member) => member.isResponsible);
  const query = `usp=pp_url&entry.1942863541=${tenant?.fullName}&entry.1090481808=${fullAddress}&entry.321634337=${propertyReference}&entry.582205322=${paymentReference}`;
  const encodedQuery = encodeURI(query);
  return encodedQuery;
};

const menu: MenuProps[] = [
  {
    label: "Sole to Joint",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform",
    processes: [
      getProcessLink(processes.soletojoint.processName),
      getProcessLink(processes.changeOfName.processName),
    ],
  },
  {
    label: "Other tenancy changes",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform",
    getPplQuery: getPplQueryTenancyChange,
  },
  {
    label: "Record a tenancy / audit homecheck",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSfyXscVvgCBhbBBe7ADpV6cUqJoKrrTbU0YYRB_uXS2J4AHMQ/viewform",
  },
  {
    label: "Find legal referral response",
    link: "https://docs.google.com/spreadsheets/d/1fd6FwBXvBo2DmX8qZaXUr7hX9sE6Zr7HZsPlA_OO8Hs/edit#gid=1251146140",
  },
  {
    label: "Request a legal referral",
    link: "https://docs.google.com/forms/d/e/1FAIpQLScpHsoJQgq112HIMbW06YAFpggijwQuODJM483HGbX_PMfCMw/viewform",
    getPplQuery: getPplQueryForRequestALegalReferral,
  },
  {
    label: "Record notice served",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSfRSMia4FeT6WPjrmRlsLpOXyJ4tFs3Sv_O0toxUbv_FmIjKw/viewform",
  },
  {
    label: "Rehousing",
    link: "https://docs.google.com/forms/d/e/1FAIpQLScPXSs4-xxTH2N5dUwtfmEl5mpu7kxZ6mHe86Qy_rlDNu-GbA/viewform",
  },
  {
    label: "Person",
    link: "https://docs.google.com/forms/d/e/1FAIpQLScxrWFLF7M7A3B1yzY-LZFXkV0P0qv6igcQ4wEgWHQk8DaUCw/viewform",
  },
  {
    label: "Report Japanese knotweed",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSceU2H-VWi2ewaEiOAkOWULVFU2DoPSen8SRPyJrmQtRpEhfQ/viewform?gxids=7628",
  },
  {
    label: "Report overgrown tree",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSfLmlLraDyOfqYhdjkOfkNWWlHUUep05G5n0GN-ewizN0Fbuw/viewform?gxids=7628",
  },
  {
    label: "Report overgrown garden",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSc3mOUUEQhdRiDsS0k9XDWlfGJG0CEFAC2rnC1sYmM-idt-Nw/viewform?gxids=7628",
  },
];

export default menu;
