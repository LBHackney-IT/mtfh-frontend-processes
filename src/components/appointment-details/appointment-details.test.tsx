import React from "react";

import { render } from "@hackney/mtfh-test-utils";
import { screen } from "@testing-library/react";

import { locale, processes } from "../../services";
import { Trigger } from "../../services/processes/types";
import {
  mockDocumentsAppointmentRescheduled,
  mockDocumentsRequestedAppointment,
} from "../../test-utils";
import { AppointmentDetails } from "./appointment-details";

const setNeedAppointment = jest.fn();
describe("appointment-details-component", () => {
  beforeEach(() => {
    jest.resetModules();
  });

  test("it renders scheduled appointment details correctly", () => {
    const { states } = processes.soletojoint;
    const { container } = render(
      <AppointmentDetails
        process={mockDocumentsRequestedAppointment({
          appointmentDateTime: "2099-10-12T08:59:00.000Z",
        })}
        needAppointment
        setNeedAppointment={setNeedAppointment}
        options={{
          requestAppointmentTrigger: Trigger.RequestDocumentsAppointment,
          rescheduleAppointmentTrigger: Trigger.RescheduleDocumentsAppointment,
          appointmentRequestedState: states.documentsRequestedAppointment.state,
          appointmentRescheduledState: states.documentsAppointmentRescheduled.state,
        }}
      />,
    );
    expect(screen.findByLabelText(locale.change)).resolves.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("it renders scheduled but missed appointment details correctly", () => {
    const { states } = processes.soletojoint;
    render(
      <AppointmentDetails
        process={mockDocumentsRequestedAppointment({
          appointmentDateTime: "2010-10-12T08:59:00.000Z",
        })}
        needAppointment
        setNeedAppointment={setNeedAppointment}
        options={{
          requestAppointmentTrigger: Trigger.RequestDocumentsAppointment,
          rescheduleAppointmentTrigger: Trigger.RescheduleDocumentsAppointment,
          appointmentRequestedState: states.documentsRequestedAppointment.state,
          appointmentRescheduledState: states.documentsAppointmentRescheduled.state,
        }}
      />,
    );
    expect(screen.findByLabelText(locale.reschedule)).resolves.toBeInTheDocument();
  });

  test("it renders rescheduled appointment details correctly", () => {
    const { states } = processes.soletojoint;
    const { container } = render(
      <AppointmentDetails
        process={{
          ...mockDocumentsAppointmentRescheduled({
            appointmentDateTime: "2099-10-12T08:59:00.000Z",
          }),
          previousStates: [
            {
              state: "DocumentsRequestedAppointment",
              permittedTriggers: [],
              assignment: "",
              processData: {
                formData: {
                  appointmentDateTime: "2010-10-12T08:59:00.000Z",
                },
                documents: [],
              },
              createdAt: "",
              updatedAt: "",
            },
            {
              state: "BreachChecksPassed",
              permittedTriggers: [],
              assignment: "",
              processData: {
                formData: {},
                documents: [],
              },
              createdAt: "",
              updatedAt: "",
            },
          ],
        }}
        needAppointment
        setNeedAppointment={setNeedAppointment}
        options={{
          requestAppointmentTrigger: Trigger.RequestDocumentsAppointment,
          rescheduleAppointmentTrigger: Trigger.RescheduleDocumentsAppointment,
          appointmentRequestedState: states.documentsRequestedAppointment.state,
          appointmentRescheduledState: states.documentsAppointmentRescheduled.state,
        }}
      />,
    );
    expect(screen.findByLabelText(locale.change)).resolves.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
