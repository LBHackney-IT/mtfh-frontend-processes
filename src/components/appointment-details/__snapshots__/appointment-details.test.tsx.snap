// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appointment-details-component it renders rescheduled appointment details correctly 1`] = `
<div>
  <div
    class="mtfh-box"
  >
    <div
      class="mtfh-status-heading"
    >
      <svg
        fill="none"
        height="45"
        viewBox="0 0 45 45"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="22.5"
          cy="22.5"
          fill="white"
          r="21.75"
          stroke="#0B0C0C"
          stroke-width="1.5"
        />
        <path
          d="M20.2251 12.7026C20.2251 11.1991 21.0627 10.4473 22.7378 10.4473C24.4129 10.4473 25.2505 11.1991 25.2505 12.7026C25.2505 13.4191 25.0386 13.9792 24.6147 14.3828C24.201 14.7764 23.5754 14.9731 22.7378 14.9731C21.0627 14.9731 20.2251 14.2163 20.2251 12.7026ZM25.0386 34H20.4219V17.0771H25.0386V34Z"
          fill="#0B0C0C"
        />
      </svg>
      <h4
        class="lbh-heading-h4 mtfh-status-heading__title"
      >
        Office appointment missed
      </h4>
    </div>
    <p
      class="lbh-body-m"
      style="margin-left: 60px;"
    >
      Date:
       
      Tuesday 12th October 2010
      <br />
      Time:
       
      08:59 am
    </p>
  </div>
  <div
    class="mtfh-box"
  >
    <div
      class="mtfh-status-heading"
    >
      <svg
        fill="none"
        height="45"
        viewBox="0 0 45 45"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="22.5"
          cy="22.5"
          fill="white"
          r="21.75"
          stroke="#0B0C0C"
          stroke-width="1.5"
        />
        <path
          d="M20.2251 12.7026C20.2251 11.1991 21.0627 10.4473 22.7378 10.4473C24.4129 10.4473 25.2505 11.1991 25.2505 12.7026C25.2505 13.4191 25.0386 13.9792 24.6147 14.3828C24.201 14.7764 23.5754 14.9731 22.7378 14.9731C21.0627 14.9731 20.2251 14.2163 20.2251 12.7026ZM25.0386 34H20.4219V17.0771H25.0386V34Z"
          fill="#0B0C0C"
        />
      </svg>
      <h4
        class="lbh-heading-h4 mtfh-status-heading__title"
      >
        Office appointment scheduled
      </h4>
    </div>
    <p
      class="lbh-body-m"
      style="margin-left: 60px;"
    >
      Date: 
      Monday 12th October 2099
      <br />
      Time: 
      08:59 am
    </p>
    <button
      class="govuk-link lbh-link"
      style="margin-left: 60px;"
      type="button"
    >
      Change
    </button>
  </div>
</div>
`;

exports[`appointment-details-component it renders scheduled appointment details correctly 1`] = `
<div>
  <div
    class="mtfh-box"
  >
    <div
      class="mtfh-status-heading"
    >
      <svg
        fill="none"
        height="45"
        viewBox="0 0 45 45"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="22.5"
          cy="22.5"
          fill="white"
          r="21.75"
          stroke="#0B0C0C"
          stroke-width="1.5"
        />
        <path
          d="M20.2251 12.7026C20.2251 11.1991 21.0627 10.4473 22.7378 10.4473C24.4129 10.4473 25.2505 11.1991 25.2505 12.7026C25.2505 13.4191 25.0386 13.9792 24.6147 14.3828C24.201 14.7764 23.5754 14.9731 22.7378 14.9731C21.0627 14.9731 20.2251 14.2163 20.2251 12.7026ZM25.0386 34H20.4219V17.0771H25.0386V34Z"
          fill="#0B0C0C"
        />
      </svg>
      <h4
        class="lbh-heading-h4 mtfh-status-heading__title"
      >
        Office appointment scheduled
      </h4>
    </div>
    <p
      class="lbh-body-m"
      style="margin-left: 60px;"
    >
      Date: 
      Monday 12th October 2099
      <br />
      Time: 
      08:59 am
    </p>
    <button
      class="govuk-link lbh-link"
      style="margin-left: 60px;"
      type="button"
    >
      Change
    </button>
  </div>
</div>
`;
